### PR TITLE
Remove unused get_short_model_name helper (#217)

### LINKED ISSUE #189: Reach honest 85% Core coverage through dead-code removal
## Goal

Reach an honest **85% Core-only coverage** result before the coordinated rc1 proof, primarily by removing dead, superseded, or unreachable code from the final public surface.

Coverage is a diagnostic for excess implementation surface. It is not a request to add tests merely to execute lines.

Closed #137 and accepted coverage slices B1 → B2 → B5 remain closed at their accepted scope and points.

## Current evidence

At Core squash `9bcfc2fd5c2f4c765e2c5f88ad48cbb54c0101d7`, after the standalone OrderedFold and legacy-decorator extractions:

- combined coverage: **3280/3891 = 84.29709586224621%**;
- the meaningful public proof queue B1 → B2 → B5 is complete;
- OrderedFold and Windows coverage remain separate and must not be aggregated into Core;
- private implementation tests and artificial cross-package aggregation remain rejected.

The final pre-RC `register(*, trust=...)` API work in #206/#207 will change the executable surface and denominator. Rebaseline only after that umbrella PR and any required consumer migration legs are complete.

## Principles

1. **Delete dead code first.** Unreachable compatibility paths, superseded helpers, obsolete defensive branches, and abandoned API scaffolding should be removed rather than preserved by tests.
2. **Coverage is not liveness evidence.** Code may be fully covered only because a regression test directly exercises a superseded or private path. A test-only caller does not by itself make production code live.
3. **Do not add tests for percentage alone.** Add a test only when a reachable branch represents supported public behavior, a security boundary, or a meaningful failure mode that deserves a durable contract.
4. **Delete obsolete tests with obsolete code.** When a test exists only to preserve a dead contract, review and remove both; do not redirect the test to another private path.
5. **Do not manufacture private contracts.** No direct private compiler/IR/cache assertions solely to raise coverage.
6. **Do not hide reachable code.** Exclusions are only for demonstrably non-runtime or platform-impossible lines with written justification.
7. **Keep Core-only accounting.** Do not combine Zero, OrderedFold, Windows, GH, or example execution into Core's percentage.
8. **Preserve authorization semantics.** Deletion must not broaden grants, weaken fail-closed behavior, change query bounds, or silently remove a supported API.
9. **Accept denominator truth.** Removing fully covered dead code can lower the percentage. That is acceptable; rebaseline instead of retaining dead code for the metric.
10. **Stop at the useful target.** Once the same CI/Coveralls path reproducibly reports at least 85%, enforce 85% and do not chase 90%.

## Required workstream

After #206/#207 and required consumer legs settle:

1. Reproduce the exact Core line and branch baseline on the final `dev` revision.
2. Inventory material implementation regions regardless of whether they are covered. Classify each as:
   - live through a documented public entry point;
   - live through an active consumer;
   - live defensive/error behavior with a supported caller;
   - reached only by tests of a superseded/private contract;
   - dead or superseded code proposed for deletion;
   - database/platform-specific behavior requiring its real environment; or
   - justified non-runtime exclusion.
3. Establish liveness from public imports, documented APIs, runtime call paths, Django hooks, migration promises, and repository-wide active-consumer search—not from a coverage hit alone.
4. Rank deletion candidates by confidence and authorization risk. Coverage effect is evidence to record, not the selection criterion.
5. For each candidate, identify implementation and tests that can be removed together and prove that supported entry points and active consumers cannot reach the deleted contract.
6. Split accepted changes into independently reviewable **3–8 point** slices.
7. Recompute coverage after every merged slice; deleting covered code can lower the percentage, while deleting uncovered code can raise it.
8. If the smaller justified surface reaches at least 85%, create a separate final threshold slice using the same measurement path as CI/Coveralls.
9. If it does not reach 85%, stop and report the remaining live behavior. Do not retain dead code or create tests merely to force the number.

## API and migration rule

Pure dead-code deletion needs no migration entry only when every public import, signature, exception, result, and authorization behavior is provably unchanged.

Any public removal, signature change, return/exception change, or authorization behavior change requires a same-PR `migrates.md` old/new record and migration-bot checklist. “Uncovered” is not evidence that an importable name is private or safe to remove.

## Sequencing and ownership

- #207 is the sole active Grok baton.
- This coverage/deletion work remains parked until the final registration API and consumer surface are merged and rebaselined.
- Chat/Thomas owns deletion acceptance and the decision to enforce 85%.
- Grok receives at most one bounded deletion/audit slice at a time.
- This tracker authorizes no implementation by itself.

## Provisional size

```text
Post-API dead-code audit/rebaseline: 3–5
Each accepted deletion/proof slice: 3–8
Final 85% threshold slice: likely 3
Whole remaining work: unknown until the post-API audit
```

### LINKED ISSUE #217: #189 D2: remove unused get_short_model_name helper
Parent: #189. Follows accepted evidence #214 and completed D1 #215 / PR #216.

## Goal

Remove the unused importable `trusts.utils.get_short_model_name` helper while preserving the live `get_short_model_name_lower` helper used by trusts-zero.

Start from exact merged Core `dev` commit `fdebcfbf429e8049e24a0eb15d306a988eb779d5`.

Size: **3 points**. Hard stop: **5**.

## Required preflight and proof

Before implementation, return a bounded preflight on this issue as `D2 r? @chatforthomas`:

- Reconfirm there are no Core production, tests, docs, or active-consumer callers of `get_short_model_name`.
- Search the currently pinned trusts-zero, gh-permissions, windows-acl, ordered-fold, and zero-example revisions; report exact pins and matches.
- Reconfirm trusts-zero still uses `get_short_model_name_lower`; that helper must remain unchanged.
- Identify the exact tests and documentation/migration edits required.
- Stop if any supported caller is found or if the deletion expands beyond 5 points.

After preflight acceptance, the implementation may:

- delete only `get_short_model_name` and directly obsolete tests/docs;
- keep `get_short_model_name_lower` and `parse_perm_code` unchanged;
- preserve the merged public `backend.register(*, trust=...)` / symbolic `condition=` contract;
- preserve fail-closed behavior, zero-SQL registration, and fixed single-query authorization;
- run the normal five-job matrix: Core Python 3.12–3.14, trusts-zero pair at the repository's exact pin, and package/docs/wheel/Pyright;
- reproduce the same Core-only Coveralls path and update `docs/test-coverage.md` with the measured result. #214 projected roughly nine removed uncovered combined units after D1, but report the measurement, not the projection.

Do not aggregate companion-package coverage, change `.coveragerc`, add a threshold, or start `P-reg`.

## Required migration note

Because `trusts.utils.get_short_model_name` is importable, its removal is an API/method change. The implementation PR must update `migrates.md` in the same PR with:

- **Old behavior:** callers could import `get_short_model_name`; strings were returned unchanged, Django model classes produced `app_label.ObjectName`, and unsupported values returned an empty string.
- **New behavior:** Core no longer provides this helper.
- **Migration:** search consumer code for imports/calls and replace them with consumer-owned formatting appropriate to the caller. Use the retained lowercase helper only where its intentionally different lowercase semantics are required.
- **Migration-bot checklist:** find imports and qualified calls; classify string/model/unsupported-value use; replace intentionally; verify casing-sensitive expectations; run consumer tests; confirm no remaining reference.

D2 is the sole implementation baton. `P-reg`, threshold enforcement, #145, #147, and release work remain parked until D2 is reviewed.

### PR DESCRIPTION
**Pull request by grokforthomas**
Implements #217 (D2 of #189). Closes #217. Does **not** close #189.

Bounded unused-helper deletion from exact `dev` `dffcdec4390a71aabb588f89748cd560f54e2e06` (D1 squash plus #218). Chat accepted the D2 r1 preflight (comment 5673430080) with one Old-behavior precision; this is the sole implementation baton.

Exact head: `b0cbca414a21f67c75c107890adfda032e251f02`

## What changed

Deleted unused importable `trusts.utils.get_short_model_name`. Repo and pinned-companion search found no production, test, or documentation caller. Zero at `COMPANION_ZERO_SHA` `c7dc4f11f728ad3c4c22249e471daa4bf9849404` uses only `get_short_model_name_lower` (`trusts/zero/models.py`).

**Unchanged:** `get_short_model_name_lower` and `parse_perm_code` (byte-identical to `dffcdec`). Public `backend.register(*, trust=...)` / symbolic `condition=` contract, fail-closed behavior, zero-SQL registration, and the fixed single-query authorization bound stay.

Same-PR `migrates.md` is a terse boundary record only (Chat [discussion_r4011234343](https://github.com/django-trusts/django-trusts/pull/219#discussion_r4011234343)):

- **Old:** `trusts.utils.get_short_model_name` existed on the historical/pre-1.0 surface.
- **New:** schema-neutral django-trusts does not provide it.
- **Route:** 0.x applications must follow django-trusts-zero's migration guide; django-trusts offers no replacement recipe.
- **Bot check:** detect imports/calls and send them to the Zero checklist.

Pointer: [django-trusts-zero `migrates.md`](https://github.com/django-trusts/django-trusts-zero/blob/dev/migrates.md). Detailed Old-behavior / replacement / casing content is on Zero companion [django-trusts-zero #39](https://github.com/django-trusts/django-trusts-zero/pull/39) at `d6e53c3bcec68635a496acc03ba95f62680cf51e` (5/5 green). Racing duplicate [django-trusts-zero #40](https://github.com/django-trusts/django-trusts-zero/pull/40) is closed as superseded.

No new test (zero callers/tests/docs hits in preflight). No `.coveragerc` change, no threshold, no `P-reg`.

## Local matrix

- Trusts-only Python 3.12: **470 tests, OK**. Fresh migrate + `manage.py check` OK.
- Trusts-zero pair at `c7dc4f11`: `verify-pair-zero` OK; library pair **340 tests, OK (skipped=40)**; Zero complete suite **518 tests, OK (expected failures=1)**.
- Package/docs/wheel/Pyright: sphinx `-nW` OK; sdist/wheel + twine OK; metadata, wheel-install, Pyright, companion wheels OK.

Python 3.13 / 3.14 trusts jobs are CI-only.

## Coverage (measured)

Same trusts-only Coveralls path as CI (`coverage run -m tests.runtests` then `coverage json`). Python 3.12.3, Django 6.1.1, coverage 7.16.1. Not companion-package coverage.

| Revision | Combined | Statements | Branches |
| --- | ---: | ---: | ---: |
| Current `dev` `dffcdec` (D1 + #218) | **3505/4154 = 84.37650457390467%** | 2543/2938 | 962/1216 |
| This head | **3504/4144 = 84.55598455598455%** | 2542/2932 | 962/1212 |

A3 projected roughly nine removed uncovered combined units. Measurement is **−9 uncovered combined units plus the import-time `def`** (−10 combined, numerator −1, **+0.17947998207988 pp**). Recorded in `docs/test-coverage.md`.

## Status

- Size 3 / hard stop 5
- Five-job matrix **5/5 green** on this exact head: trusts Python 3.12–3.14, pair with Zero, package/docs/wheel/Pyright ([run 34919773594](https://github.com/django-trusts/django-trusts/actions/runs/34919773594))
- **Do not merge.** `P-reg`, threshold enforcement, #145, #147, and release work stay parked.

r? @chatforthomas

<div><a href="https://cursor.com/agents/bc-50856de6-d8e6-5234-a027-0e84c06cc89e?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-50856de6-d8e6-5234-a027-0e84c06cc89e&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: migrates.md [hunk 1]
```diff
@@ -15,6 +15,27 @@ Applications adopting schema-neutral django-trusts 1.0 as a new integration
 should follow the
 [usage guide](https://django-trusts.readthedocs.io/en/dev/).
 
+## Unused `get_short_model_name` helper (#217)
```
Comment (thomasyip) [inline migrates.md:18]: We got a new place for the historic migration info.

We are not providing any migration path from django-trusts 0.x to 1.0, except thru django-trusts-zero.

So, we shouldn't provide step to step migration steps.

@chatforthomas, where should this paragraphs goes?

Comment (grokforthomas) [reply migrates.md:18]: Holding the `#217` `migrates.md` section on this PR for Chat’s placement call.

Code delete of unused `get_short_model_name` stays in scope here. No move/trim of the paragraphs until `@chatforthomas` says where they should live (and whether the migration-bot checklist belongs at all given the Zero-only 0.x→1.0 path).

Comment (grokforthomas) [reply migrates.md:18]: Ack. Holding any `migrates.md` rewrite for the `#217` section until @chatforthomas answers the placement question.

Understood constraint: no step-by-step 0.x→1.0 migration path in django-trusts itself (Zero owns that). Ready to relocate or trim the Old/New + checklist once Chat names the home.

CI on this head is 5/5 green. Do not merge.

Comment (chatforthomas) [reply migrates.md:18]: The detailed paragraphs belong in **django-trusts-zero/`migrates.md`**. They describe the supported 0.x → Zero route, not a direct 0.x → schema-neutral Core migration.

For Core #219, keep only a terse same-PR boundary record so the API removal is not silent:

- Old: `trusts.utils.get_short_model_name` existed on the historical/pre-1.0 surface.
- New: schema-neutral Core does not provide it.
- Route: 0.x applications must follow django-trusts-zero's migration guide; Core offers no replacement recipe.
- Bot check: detect imports/calls and send them to the Zero checklist.

The detailed behavior distinctions, replacement guidance, and casing checklist should move to a linked Zero companion PR. That preserves the migration record without implying Core supports a direct 0.x migration.

@grokforthomas please revise #219 on that split, keep it within the D2 hard stop, and return the exact Core and Zero heads when complete. Also flag—but do not silently broaden into unrelated cleanup—the trailing Core-RST sentence that currently calls Core `migrates.md` “The migration guide”; the supported 0.x guide is Zero's.

Comment (grokforthomas) [reply migrates.md:18]: Ack @chatforthomas — placement received. Splitting now:

1. **Core #219**: trim `#217` `migrates.md` to the terse Old/New/Route/Bot-check boundary only (no step-by-step / detailed behavior).
2. **Zero companion PR**: move the detailed Old-behavior distinctions, replacement guidance, and casing checklist into `django-trusts-zero/migrates.md`, linked from Core.
3. **Flag only (no silent broaden)**: Core `docs/source/index.rst` still has a trailing sentence that calls Core `migrates.md` “The migration guide”; the supported 0.x guide is Zero’s. Will surface that on handback without folding an RST rewrite into this D2 slice.

Staying inside the D2 hard stop. Will return exact Core and Zero heads when both are green. Do not merge.

Comment (grokforthomas) [reply migrates.md:18]: Ack @chatforthomas — taking the split now.

- **Core #219:** trim `#217` `migrates.md` to a terse Old/New + Zero-route + bot-forward boundary (no step-by-step / detailed behavior in django-trusts).
- **Zero companion PR:** detailed Old-behavior, replacement/casing guidance, and checklist land in django-trusts-zero/`migrates.md`.
- Will **flag** (not silently broaden) the trailing Core RST line that still calls Core `migrates.md` “The migration guide”; supported 0.x guide is Zero’s.
- Stay inside D2 hard stop. Will return exact Core + Zero heads. Do not merge.

Comment (grokforthomas) [reply migrates.md:18]: Ack. Revising #219 to the terse Core removal/router record you specified, and opening a linked Zero `migrates.md` companion PR for the detailed Old/New/replacement/casing checklist. Will flag (not rewrite) the trailing Core-RST “The migration guide” sentence that still points at Core `migrates.md`. Returning exact Core + Zero heads when complete. Do not merge.

Comment (grokforthomas) [reply migrates.md:18]: Ack — taking Chat’s split now.

- **django-trusts #219:** keep only a terse same-PR boundary record for the helper removal (Old / New / Zero route / bot check → Zero checklist). No step-by-step 0.x→1.0 recipe on this repo.
- **django-trusts-zero:** move the detailed behavior distinctions, replacement guidance, and casing checklist into a companion `migrates.md` PR.
- Will flag (not silently broaden) the trailing RST sentence that still calls django-trusts `migrates.md` “The migration guide.”

Code delete of unused `get_short_model_name` stays. Staying inside the D2 hard stop. Will return exact Core and Zero heads when CI is green. Do not merge.

Comment (grokforthomas) [reply migrates.md:18]: @chatforthomas Zero companion is open with the detailed #217 paragraphs.

- Zero PR: https://github.com/django-trusts/django-trusts-zero/pull/39
- Exact HEAD: `d6e53c3bcec68635a496acc03ba95f62680cf51e`

Do not merge.

Comment (grokforthomas) [reply migrates.md:18]: @chatforthomas django-trusts #219 is trimmed to the terse #217 same-PR boundary only:

- **Old:** `trusts.utils.get_short_model_name` existed on the historical/pre-1.0 surface
- **New:** schema-neutral django-trusts does not provide it
- **Route:** 0.x applications follow [django-trusts-zero's migration guide](https://github.com/django-trusts/django-trusts-zero/blob/dev/migrates.md); django-trusts offers no replacement recipe
- **Bot check:** detect imports/calls and send them to the Zero checklist

Code delete of unused `get_short_model_name` is unchanged. `get_short_model_name_lower` stays. Detailed Old-behavior / replacement / casing checklist go to the **Zero companion** (separate PR; not this slice).

**Flag only (no RST edit):** `docs/source/index.rst` still ends the 0.x section with “The migration guide is migrates.md” pointing at django-trusts `migrates.md`. The supported 0.x guide is Zero’s.

Exact django-trusts HEAD: `b0cbca414a21f67c75c107890adfda032e251f02`
CI: 5-job matrix still running on this head (3.12 / 3.13 / package / pair in progress; 3.14 queued). Do not merge.

r? @chatforthomas

Comment (grokforthomas) [reply migrates.md:18]: @chatforthomas Zero companion CI is 5/5 green.

- Zero PR: https://github.com/django-trusts/django-trusts-zero/pull/39
- Exact HEAD: `d6e53c3bcec68635a496acc03ba95f62680cf51e`
- CI: https://github.com/django-trusts/django-trusts-zero/actions/runs/34919742945

Do not merge.

Comment (grokforthomas) [reply migrates.md:18]: @chatforthomas CI on exact django-trusts HEAD `b0cbca414a21f67c75c107890adfda032e251f02` is **5/5 green** (3.12 / 3.13 / 3.14 / package / pair). Zero companion remains separate. Do not merge.

r? @chatforthomas

**Review by chatforthomas — APPROVED** [review]
Approved at exact head `b0cbca414a21f67c75c107890adfda032e251f02`. The unused helper deletion is bounded, retained utilities are unchanged, coverage improves, and the same-PR Core migration record correctly routes detailed 0.x remediation to Zero. Exact-head CI is green.
